src/Catalog/Monitoring/Validation/Test/ValidationTestRunnerFactory.cs

```diff
@@ -0,0 +1,111 @@
+﻿using NuGet.Protocol.Core.Types;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Protocol;
+using System.Reflection;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test
+{
+    /// <summary>
+    /// Creates <see cref="IValidationTestRunner"/>s using <see cref="SourceRepository"/>s.
+    /// </summary>
+    public class ValidationTestRunnerFactory
+    {
+        /// <summary>
+        /// Exception wrapper for unexpected exceptions thrown during initialization of <see cref="IValidationTestRunner"/>s.
+        /// </summary>
+        public class ValidationTestRunnerFactoryException : Exception
+        {
+            private Type _validationTestRunnerType;
+            private IDictionary<ConstructorInfo, Exception> _exceptionsForConstructor;
+
+            public ValidationTestRunnerFactoryException(Type validationTestType, IDictionary<ConstructorInfo, Exception> exceptionsForConstructor)
+            {
+                _exceptionsForConstructor = exceptionsForConstructor;
+                _validationTestRunnerType = validationTestType;
+            }
+
+            public override string Message
+            {
+                get
+                {
+                    var message = $"Could not initialize {_validationTestRunnerType.Name}.";
+
+                    var constructorMessages = _exceptionsForConstructor.Select(pair => $"Constructor with parameters {string.Join(", ", pair.Key.GetParameters().Select(p => p.ParameterType.Name))} failed with exception: {pair.Value}");
+                    message += (constructorMessages.Any() ? " " : "") + string.Join(", ", constructorMessages);
+
+                    return message;
+                }
+            }
+        }
+        
+        /// <param name="feedToSource">Used to map <see cref="IResourceType"/> to the <see cref="SourceRepository"/> to use.</param>
+        public ValidationTestRunnerFactory(IDictionary<FeedType, SourceRepository> feedToSource)
+        {
+            _feedToSource = feedToSource;
+        }
+
+        public IValidationTestRunner Create(Type validationType)
```

Comment:
>Create [](start = 37, length = 6)

Yikes! this code is pretty gnarly. please consider hard coding construction to avoid all these reflection (which is a lot harder to debug).